examples/gno.land/r/demo/teritori/social/user.gno

```diff
@@ -0,0 +1,58 @@
+package social
+
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+)
+
+type User struct {
+	address std.Address
+	followers avl.Tree
+	followeds avl.Tree
+}
+
+func (u *User) Address() std.Address {
+	return u.address
+}
+
+func (u *User) Followers() []std.Address {
+	followers := make([]std.Address, 0, u.followers.Size())
+	u.followers.Iterate("", "",  func(key string, value interface{}) bool  {
+		follower := value.(*User)
+		followers = append(followers, follower.address)
+		return false
+	})
+	return followers
+}
+
+func (u *User) Followed() []std.Address {
+	followeds := make([]std.Address, 0, u.followeds.Size())
+	u.followeds.Iterate("", "",  func(key string, value interface{}) bool  {
+		followed := value.(*User)
+		followeds = append(followeds, followed.address)
+		return false
+	})
+	return followeds
+}
+
+func (u *User) Follow(user *User) {
+	if _, ok := u.followeds.Get(user.address.String()); ok {
```

Comment:
question: should we allow user to follow himself ?

Reply:
good question, I don't think so, do you see a potential usecase for it?